distributed/tests/test_worker_plugins.py

```diff
@@ -18,6 +23,37 @@ def teardown(self, worker):
         assert isinstance(worker, Worker)
         self.worker._my_plugin_status = "teardown"
 
+    def task_started(self, worker, key, args, kwargs):
+        assert isinstance(worker, Worker)
+
+        if len(self.expected_args) > 0:
+            _args, _kwargs = self.expected_args.pop(0)
+            if _args is not None:
+                assert args == _args
+            if _kwargs is not None:
+                assert kwargs == _kwargs
+
+    def task_finished(self, worker, key, result):
+        assert isinstance(worker, Worker)
+
+        if len(self.expected_results) > 0:
+            assert result == self.expected_results.pop(0)
+
+    def task_failed(self, worker, key, exception):
```

Comment:
would be helpful to have the failed callback somewhat aligned with the Python `__exit__` functions, so we get `type` (exception type), `value` (exception value) and `tb` (traceback). See [Python docs](https://docs.python.org/3/library/stdtypes.html#typecontextmanager) as well. A use case would be that the handler could than analyze/report the traceback.

Reply:
that's a good idea, thanks!